### PR TITLE
add viewer

### PR DESCRIPTION
**Pull request by vincentsarago**
ref #27 
Screenshot 2023-02-21 at 12 40 17 AM

### CODE REVIEW

File: tipg/factory.py [hunk 1]
```diff
@@ -1188,7 +1188,7 @@ async def tile(
             return Response(bytes(tile), media_type=MediaType.mvt.value)
 
         @self.router.get(
-            "/collections/{collectionId}/{TileMatrixSetId}/tilejson.json",
+            "/collections/{collectionId}/{tileMatrixSetId}/tilejson.json",
```
Comment (vincentsarago) [inline tipg/factory.py:1191]: should have added a test for this BUG

File: tipg/templates/map.html [hunk 2]
```diff
@@ -0,0 +1,159 @@
+<!DOCTYPE html>
```
Comment (vincentsarago) [inline tipg/templates/map.html:1]: This was the timvt viewer

File: tipg/factory.py [hunk 3]
```diff
@@ -1251,17 +1251,72 @@ async def tilejson(
...
+        @self.router.get(
+            "/collections/{collectionId}/viewer", response_class=HTMLResponse
+        )
+        def viewer_endpoint(
+            request: Request,
+            collection=Depends(self.collection_dependency),
+            tileMatrixSetId: Literal["WebMercatorQuad"] = Query(  # noqa
+                "WebMercatorQuad",
+                description="Identifier selecting one of the TileMatrixSetId supported (default: 'WebMercatorQuad')",
+            ),
```
Comment (vincentsarago) [inline tipg/factory.py:1284]: I choose to use the timvt viewer (using Maplibre) which only works with WebMercatorQuad TMS. The main issue is if a user will implement endpoints without WebMercatorQuad in the `supported_tms` 🤷

if we wanted to use leaflet + support multiple TMS we could use https://gist.github.com/vincentsarago/2fe69adda0b5089e9e6ca6bfe99bc7c9

Comment (bitner) [reply tipg/factory.py:1284]: I think maplibre is just fine. This viewer is mostly for quick testing/verifying.

**Review by bitner — APPROVED** [review]

**Pull request merged**